File: 8_remove-duplicates-from-sorted-array/solution.py
def removeDuplicates(nums):
    """
    :type nums: List[int]
    :rtype: int
    """
    if not nums: 
        return 0
    
    counter = 0
    
    for i in range(1, len(nums)):
        print(i)
        if nums[i] != nums[counter]:  
            counter += 1  
            nums[counter] = nums[i] 
    
    return counter + 1 , nums

nums = [0,0,1,1,1,2,2,3,3,4]

counter, return_list = removeDuplicates(nums)

print( counter ,return_list)

File: 8_remove-duplicates-from-sorted-array/solution.cpp
#include <vector>
#include <iostream>
int removeDuplicates(std::vector<int>& nums) {
    int counter = 0;
    for(int i = 1; i < nums.size();i++) {
        if (nums.at(i) != nums.at(counter)) {
            counter++;
            nums.at(counter) = nums.at(i);
        }
    }

    return counter+1;
}
int main(){
    std::vector<int> nums = {1,2,3,3,3,4,5,6,7};
    int count = removeDuplicates(nums); 
    // std::cout << nums.at(0) << std::endl; 
    return 0;
}